### PR TITLE
Fix panic in comparison_kernel benchmarks

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-rs/issues/6283

# Rationale for this change
 
Benchmark was panic'ing

Also having benchmarks in main ahead of  a proposed code change makes it easier for my scripts to compare performance before/after code change

# What changes are included in this PR?

Also, adds benchmarks for code in https://github.com/apache/arrow-rs/pull/6231 to make it easier to verify performance.

# Are there any user-facing changes?
No, this is development tooling only

### CODE REVIEW

File: arrow/benches/comparison_kernels.rs [hunk 1]
```diff
@@ -240,10 +241,18 @@ fn add_benchmark(c: &mut Criterion) {
...
-    c.bench_function("like_utf8view scalar starts with", |b| {
+    c.bench_function("like_utf8view scalar starts with less than 4 bytes", |b| {
         b.iter(|| bench_like_utf8view_scalar(&string_view_left, "xxxx%"))
     });
 
+    c.bench_function("like_utf8view scalar starts with more than 4 bytes", |b| {
+        b.iter(|| bench_like_utf8view_scalar(&string_view_left, "xxxxxx%"))
+    });
+
+    c.bench_function("like_utf8view scalar starts with more than 12 bytes", |b| {
```
Comment (alamb) [inline arrow/benches/comparison_kernels.rs:252]: I also added a second function for more than 12 bytes

**Comment by alamb** [comment]
FYI @xinlifoobar

File: arrow/benches/comparison_kernels.rs [hunk 2]
```diff
@@ -192,8 +192,20 @@ fn add_benchmark(c: &mut Criterion) {
         })
     });
 
-    c.bench_function("eq scalar StringViewArray", |b| {
-        b.iter(|| eq(&scalar, &string_view_left).unwrap())
```
Comment (alamb) [inline arrow/benches/comparison_kernels.rs:196]: the panic came from this line which passes a scalar of type `StringArray` to a comparison kernel with `StringViewArray` -- the fix was to make the appropriate scalar type with `let string_view_scalar = StringViewArray::new_scalar("xxxx");`

**Review by appletreeisyellow — APPROVED** [review]
Makes sense to me! Thank you @alamb!

**Review by wiedld — APPROVED** [review]
Confirmed I got the error (on master) when running `cargo bench --profile=dev --bench comparison_kernels`. Confirmed this change removes the error.

**Comment by alamb** [comment]
Thank you @appletreeisyellow  and @wiedld for the reviews

**Pull request merged**